shell/platform/darwin/macos/framework/Source/FlutterMenuPlugin.mm

```diff
@@ -0,0 +1,420 @@
+    }
+    if ([item hasSubmenu]) {
+      [self reapplyAppName:[[item submenu] itemArray]];
+    }
+  }
+}
+
+/**
+ * Removes all but one top-level menu from the app menu, leaving the last one
+ * with no menu items.
+ */
+- (void)removeAllMenus {
+  if ([NSApp.mainMenu numberOfItems] == 0) {
+    // There should always be at least one item to represent the APP_NAME menu.
+    [NSApp.mainMenu addItem:[[NSMenuItem alloc] init]];
+  }
+  // Clean out any existing menu items. We can't just call removeAllItems
+  // because that destroys the <APP_NAME> menu, which causes an  button (in
+  // addition to the usual  logo system menu) to be put into the menu as a
+  // placeholder, and it can't be removed once it appears. So, we remove all the
+  // menus at index 1+, and then all the items in the menu at index 0. When we
+  // add our first menu, we skip creating a new one, and just use the existing
+  // one.
+  while ([NSApp.mainMenu numberOfItems] > 1) {
+    [NSApp.mainMenu removeItemAtIndex:1];
+  }
+  NSMenu* appNameMenu = [[NSApp.mainMenu itemArray][0] submenu];
+  while ([appNameMenu numberOfItems] > 0) {
+    [appNameMenu removeItemAtIndex:0];
+  }
+  [_menuDelegates removeAllObjects];
+}
+
+/**
+ * Removes Flutter menu items that were previously added, then builds and adds
+ * new top-level menu items constructed from |representation|.
+ */
+- (void)setMenu:(NSArray*)representation {
+  [self removeAllMenus];
+  bool firstMenu = true;
+  int index = 0;
+  for (NSDictionary* item in representation.objectEnumerator) {
+    NSMenuItem* menuItem = [self menuItemFromFlutterRepresentation:item];
+    menuItem.representedObject = self;
+    if (firstMenu) {
+      // If we're adding the first menu in the menu bar, then just set the items
+      // in the submenu, rather than adding a new menu. This is because AppKit
+      // gets very squirrelly when you remove the <APP_NAME> menu, so we just
+      // empty it out in removeAllMenus, and refill it here. This ignores the
+      // label of the first item (because it will always be set to the APP_NAME
+      // by AppKit).
+      NSMenu* appNameMenu = [[NSApp.mainMenu itemAtIndex:0] submenu];
+      for (int i = 0; i < [menuItem.submenu numberOfItems]; ++i) {
+        [appNameMenu addItem:[[menuItem.submenu itemAtIndex:i] copy]];
+      }
+      firstMenu = false;
+    } else {
+      [NSApp.mainMenu addItem:menuItem];
+    }
+    NSNumber* boxedID = item[kIdKey];
+    FlutterMenuDelegate* delegate = [[FlutterMenuDelegate alloc] initWithId:boxedID.longLongValue
+                                                                 andChannel:_channel];
+    [_menuDelegates addObject:delegate];
+    [[NSApp.mainMenu itemAtIndex:index++] submenu].delegate = delegate;
+  }
+}
+
+- (NSMenuItem*)findDefaultMenu:(NSMenu*)menu ofType:(SEL)selector withTag:(int)tag {
+  auto enumerator = menu != nullptr ? menu.itemArray.objectEnumerator
```

Comment:
Please don't use `auto` for ObjC types. Also, you don't need an enumerator to use fast enumeration syntax; this can just be the NSArray itself.

Reply:
OK, done.